client/ladder/action-creators.ts

```diff
@@ -3,33 +3,52 @@ import { MatchmakingType } from '../../common/matchmaking'
 import { apiUrl, urlPath } from '../../common/urls'
 import { ThunkAction } from '../dispatch-registry'
 import { push } from '../navigation/routing'
+import { abortableThunk, RequestHandlingSpec } from '../network/abortable-thunk'
 import { fetchJson } from '../network/fetch'
 
 const LADDER_RANKINGS_CACHE_TIME_MS = 60 * 1000
 
-export function getRankings(matchmakingType: MatchmakingType): ThunkAction {
-  return (dispatch, getState) => {
+const lastFetchTimeByMatchmakingType = new Map<MatchmakingType, Date>()
```

Comment:
Using Dates for anything like this is a mistake and will not function correctly across time changes. Use a monotonic counter (`performance.now()`)